include/Types.h

```diff
@@ -370,5 +370,11 @@ std::ostream&>::type operator<<(std::ostream& os, T val)
 	return os;
 }
 
+/* Comment Type */
+typedef struct {
```

Comment:
Why should this be a `typedef` for an unnamed struct? Why not simply `struct Comment {`?

Reply:
Damn sorry, I am used to using struct with typedef so that i don't have to write struct in front of every parameter where i am using it but i forgot that i don't have to do this with c++ anymore.